synfig-studio/src/gui/widgets/widget_link.cpp

```diff
@@ -79,37 +137,61 @@ Widget_Link::Widget_Link(const std::string &tlt_inactive, const std::string &tlt
 
 	icon_off_->show();
 	add(*icon_off_);
-	set_relief(Gtk::RELIEF_NONE);
-
-	tooltip_inactive_ = tlt_inactive;
-	tooltip_active_ = tlt_active;
-	set_tooltip_text(tooltip_inactive_);
 }
 
-Widget_Link::~Widget_Link() {
-	delete icon_on_;
-	delete icon_off_;
+void
+Widget_Link::on_toggled()
+{
+	Image *icon;
+
+	// refresh icon
+	get_active() ? icon = icon_on_ : icon = icon_off_;
+
+	// refresh tooltip text
+	if (IS_WIDGET_FROM_BUILDER)
+		refresh_tooltip_text();
+	else
+		get_active() ? set_tooltip_text(tooltip_active_) : set_tooltip_text(tooltip_inactive_);
+
+	remove();
+	add(*icon);
+	icon->show();
 }
 
+void
+Widget_Link::refresh_tooltip_text()
+{
+	if (get_active())
+		set_tooltip_text(property_tooltip_active().get_value());
+	else
+		set_tooltip_text(property_tooltip_inactive().get_value());
```

Comment:
IIRC you don't need the `get_value()`

Reply:
Okay :)

Reply:
@rodolforg we'd have to cast it though otherwise it won't work
```c++
set_tooltip_text((string)property_tooltip_active());
```
I found a file `synfig-studio/src/gui/trees/keyframetree.cpp` that does exactly what you mentioned about not needing `get_value()`
https://github.com/synfig/synfig/blob/b2773121ccd27694b8b86f01e896db25647bef56/synfig-studio/src/gui/trees/keyframetree.cpp#L78
That's possible because the function `add_attribute()` can take in `Glib::PropertyProxy_Base` as argument. Sadly our `set_tooltip_text()` only takes in string. If we don't cast our custom property, synfig complains while building.

I think the current code is much better than casting. I don't want to change it. Your opinion..?